dotnet/src/Connectors/VectorData.Abstractions/RecordAttributes/VectorStoreRecordKeyAttribute.cs

```diff
@@ -18,4 +18,17 @@ public sealed class VectorStoreRecordKeyAttribute : Attribute
     /// E.g. the property name might be "MyProperty" but the storage name might be "my_property".
     /// </summary>
     public string? StoragePropertyName { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the key should be auto-generated by the vector store.
+    /// </summary>
+    /// <value>
+    /// The default is <see langword="false" />.
+    /// </value>
+    /// <remarks>
+    /// If set to <see langword="false" />, you must set the key property on any record that you pass to <see cref="IVectorStoreRecordCollection{TKey, TRecord}.UpsertAsync(TRecord, System.Threading.CancellationToken)"/>.
+    /// If set to <see langword="true" />, the key property may be left null on any record that you pass to <see cref="IVectorStoreRecordCollection{TKey, TRecord}.UpsertAsync(TRecord, System.Threading.CancellationToken)"/>
+    /// and a generated key will be returned.
+    /// </remarks>
+    public bool AutoGenerate { get; set; }
```

Comment:
I'm not sure it's right to add this to the abstraction (see the discussion in #10692); in a nutshell, AFAICT pure vector databases do not support database-generation of keys, so this may be a general relational capability that doesn't necessarily make sense in a vector database abstraction.

In any case, it seems that this is only implemented for SQL Server, and that the PG and SQLite providers simply don't support auto-generated keys at the moment... We should probably have a discussion on what we want to do on this, and comment this out for now or something...).